### ISSUE TITLE
Incorrect padding on hexadecimal formatting of integers

### ISSUE THREAD

**Issue by coder543** [rust-lang/rust#50621]
The behavior in https://github.com/rust-lang/rust/pull/48978 does not seem correct.

[Playground link](https://play.rust-lang.org/?gist=ccadba488e9120ff29235cbdfdca799b&version=beta&mode=debug)

The number of digits in the output doesn't seem to match the requested amount of padding.

**Comment by coder543** [comment]
It appears to be caused by the leading `0x`, as demonstrated by it working intuitively on [this playground.](https://play.rust-lang.org/?gist=81f20b1dec2557fca4c90e922e041899&version=beta&mode=debug)

Maybe this is expected behavior to some people, but it certainly isn't what I was expecting.

**Comment by cuviper** [comment]
You get the same thing as printing numbers directly with `{:#08x}`, which also does count the `0x`.

**Comment by fpgaminer** [comment]
Yes, this is the previous behavior when printing numbers directly.  It includes the prefix as part of the length of the formatted number.  It's odd behavior, but not new (I hit the paper cut previously ... it really is weird behavior, but I don't think we can change it now).
